From 6924b49bf425806ce1c680c283f7e62b8251ae57 Mon Sep 17 00:00:00 2001
From: Raymond Wong <infiwang@pm.me>
Date: Thu, 30 Mar 2023 00:37:41 +0800
Subject: [PATCH 074/156] riscv(misc): revert early development workarounds in
 Makefile

- "CCOPT_riscv64=-march=rv64g": DynASM do not support RVC, attempt to stop RVC
  instruction from generating in "lj_vm.o".'
  This should be handled in "host/buildvm_asm.c"

- "BUILDMODE=static": workaround for linking without pseudo GOT table

This is a WIP, archive only.
---
 src/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 1eefde6f..108825e4 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -53,7 +53,7 @@ CCOPT_arm=
 CCOPT_arm64=
 CCOPT_ppc=
 CCOPT_mips=
-CCOPT_riscv64= -march=rv64g
+CCOPT_riscv64=
 #
 CCDEBUG=
 # Uncomment the next line to generate debug information:
@@ -72,10 +72,10 @@ CCWARN= -Wall
 # as dynamic mode.
 #
 # Mixed mode creates a static + dynamic library and a statically linked luajit.
-#BUILDMODE= mixed
+BUILDMODE= mixed
 #
 # Static mode creates a static library and a statically linked luajit.
-BUILDMODE= static
+#BUILDMODE= static
 #
 # Dynamic mode creates a dynamic library and a dynamically linked luajit.
 # Note: this executable will only run when the library is installed!
-- 
2.42.0

